src/View/Helper/TimeHelper.php

```diff
@@ -333,6 +333,10 @@ public function i18nFormat($date, $format = null, $invalid = false, $timezone =
             return $invalid;
         }
 
+        if ($date instanceof \DateTimeImmutable) {
+            $date = $date->toMutable();
+        }
+
```

Comment:
I think this fix should actually happen inside Time class:
https://github.com/cakephp/cakephp/blob/master/src/I18n/Time.php#L105

Here it is modified even if immutable and should be cloned before, instead.